runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -39,14 +44,98 @@ pub(crate) struct PartitionedStakeReward {
     pub commission: u8,
 }
 
-type PartitionedStakeRewards = Vec<PartitionedStakeReward>;
+/// A vector of stake rewards.
+#[derive(Debug, Default)]
+pub(crate) struct PartitionedStakeRewards {
+    /// Inner vector.
+    rewards: Vec<Option<PartitionedStakeReward>>,
+    /// Number of `Some` elements.
+    len_some: usize,
+}
+
+impl PartitionedStakeRewards {
+    pub(crate) fn with_capacity(capacity: usize) -> Self {
+        let rewards = Vec::with_capacity(capacity);
+        Self {
+            rewards,
+            len_some: 0,
+        }
+    }
+
+    pub(crate) fn capacity(&self) -> usize {
+        self.rewards.capacity()
+    }
+
+    /// Number of `Some` elements.
+    pub(crate) fn len_some(&self) -> usize {
+        self.len_some
+    }
+
+    pub(crate) fn get(&self, index: usize) -> Option<&Option<PartitionedStakeReward>> {
+        self.rewards.get(index)
+    }
+
+    pub(crate) fn iter(&self) -> Iter<'_, Option<PartitionedStakeReward>> {
+        self.into_iter()
+    }
+
+    fn spare_capacity_mut(&mut self) -> &mut [MaybeUninit<Option<PartitionedStakeReward>>] {
+        self.rewards.spare_capacity_mut()
+    }
+
+    unsafe fn set_len(&mut self, new_len: usize) {
+        self.rewards.set_len(new_len)
+    }
```

Comment:
This is only called with `self.capacity()` so how about changing this method to something like

```
unsafe fn assume_init(&mut self) {
  self.rewards.set_len(self.rewards.capacity());
}
```

And then remove the `fn capacity` getter